Flaky-test: BrokerServiceLookupTest.testModularLoadManagerSplitBundle

BrokerServiceLookupTest is flaky. The testModularLoadManagerSplitBundle test method fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/4397027419?check_suite_focus=true#step:9:587)

```
Error:  Tests run: 25, Failures: 6, Errors: 0, Skipped: 18, Time elapsed: 67.812 s <<< FAILURE! - in org.apache.pulsar.client.api.BrokerServiceLookupTest
Error:  testModularLoadManagerSplitBundle(org.apache.pulsar.client.api.BrokerServiceLookupTest)  Time elapsed: 14.315 s  <<< FAILURE!
org.awaitility.core.ConditionTimeoutException: Assertion condition defined as a lambda expression in org.apache.pulsar.client.api.BrokerServiceLookupTest that uses org.apache.pulsar.broker.PulsarService, org.apache.pulsar.broker.PulsarServiceorg.apache.pulsar.common.naming.TopicName, org.apache.pulsar.common.naming.TopicNameorg.apache.pulsar.common.naming.NamespaceBundle did not expect [my-property/my-ns/0x00000000_0xffffffff] but found [my-property/my-ns/0x00000000_0xffffffff] within 10 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at org.apache.pulsar.client.api.BrokerServiceLookupTest.testModularLoadManagerSplitBundle(BrokerServiceLookupTest.java:741)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.AssertionError: did not expect [my-property/my-ns/0x00000000_0xffffffff] but found [my-property/my-ns/0x00000000_0xffffffff]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failEquals(Assert.java:1041)
	at org.testng.Assert.assertNotEqualsImpl(Assert.java:147)
	at org.testng.Assert.assertNotEquals(Assert.java:1531)
	at org.testng.Assert.assertNotEquals(Assert.java:1535)
	at org.apache.pulsar.client.api.BrokerServiceLookupTest.lambda$testModularLoadManagerSplitBundle$5(BrokerServiceLookupTest.java:742)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	... 4 more
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/BrokerServiceLookupTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/BrokerServiceLookupTest.java
@@ -654,14 +654,17 @@ public class BrokerServiceLookupTest extends ProducerConsumerBase {
             conf2.setLoadManagerClassName(ModularLoadManagerImpl.class.getName());
             conf2.setZookeeperServers("localhost:2181");
             conf2.setConfigurationStoreServers("localhost:3181");
-
-            @Cleanup
-            PulsarService pulsar2 = startBroker(conf2);
+            conf2.setLoadBalancerAutoBundleSplitEnabled(true);
+            conf2.setLoadBalancerAutoUnloadSplitBundlesEnabled(true);
+            conf2.setLoadBalancerNamespaceBundleMaxTopics(1);
 
             // configure broker-1 with ModularLoadManager
             stopBroker();
             conf.setLoadManagerClassName(ModularLoadManagerImpl.class.getName());
             startBroker();
+
+            @Cleanup
+            PulsarService pulsar2 = startBroker(conf2);
 
             pulsar.getLoadManager().get().writeLoadReportOnZookeeper();
             pulsar2.getLoadManager().get().writeLoadReportOnZookeeper();
@@ -732,9 +735,6 @@ public class BrokerServiceLookupTest extends ProducerConsumerBase {
                     .getLoadManager().get()).getLoadManager();
 
             updateAllMethod.invoke(loadManager);
-            conf2.setLoadBalancerAutoBundleSplitEnabled(true);
-            conf2.setLoadBalancerAutoUnloadSplitBundlesEnabled(true);
-            conf2.setLoadBalancerNamespaceBundleMaxTopics(1);
             loadManager.checkNamespaceBundleSplit();
 
             // (6) Broker-2 should get the watch and update bundle cache

